src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/MessagePackHubProtocolTest.java

```diff
@@ -0,0 +1,914 @@
+package com.microsoft.signalr;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.lang.reflect.Type;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import okio.ByteString;
```

Comment:
Why is ByteString being used?

Reply:
I found it easier to compare ByteStrings than byte[]/ByteBuffers. I could always just write a utility method to compare two byte[]s and get rid of this

Reply:
Actually I remember, it's because if you do assertTrue(Arrays.equal(byte[], byte[])), the error message is just `excepted true but got false`. AssertEquals gives a better message

Reply:
The issue is okhttp wont be referenced when this is moved to a different package

Reply:
ah, right, I'll just make my own version